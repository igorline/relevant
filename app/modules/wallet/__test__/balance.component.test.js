import React from 'react';
import renderer from 'react-test-renderer';
import { Balance } from 'modules/wallet/balance.component';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';

jest.mock('react-redux', () => {
  const { user } = require('app/mockdata');
  const state = {
    auth: { user: user.user1 },
    navigation: { screenSize: 0 }
  };

  return {
    useDispatch: () => {},
    useSelector: fn => fn(state),
    connect: el => el
  };
});

const props = {
  isWeb: true
};

test('Snapshot Balance Web', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Balance {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
