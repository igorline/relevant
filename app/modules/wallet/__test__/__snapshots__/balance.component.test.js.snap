// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Balance Web 1`] = `
.c0 {
  color: #0000ff;
}

<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "display": "flex",
      "marginBottom": "1rem",
      "marginLeft": "2rem",
      "marginRight": "2rem",
      "marginTop": "0rem",
      "position": "relative",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "display": "flex",
        "position": "relative",
      }
    }
  >
    <div
      className="css-text-901oao"
      dir="auto"
      style={
        Object {
          "color": "rgba(0,0,0,1.00)",
          "fontFamily": "HelveticaNeue",
          "fontSize": "0.875rem",
          "lineHeight": "1.2249999999999999rem",
          "marginTop": "1rem",
        }
      }
    >
      These are coins you earned as rewards. You can transfer up to 
      1000
      
       coins to your Ethereum wallet (this limit will be increased as the network grows).
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomStyle": "solid",
        "borderLeftColor": "rgba(222,222,222,1.00)",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "1px",
        "borderRightColor": "rgba(222,222,222,1.00)",
        "borderRightStyle": "solid",
        "borderRightWidth": "1px",
        "borderTopColor": "rgba(222,222,222,1.00)",
        "borderTopStyle": "solid",
        "borderTopWidth": "1px",
        "display": "flex",
        "marginTop": "1rem",
        "paddingBottom": "1rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "1rem",
        "position": "relative",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitBoxDirection": "normal",
          "WebkitBoxLines": "multiple",
          "WebkitBoxOrient": "horizontal",
          "WebkitBoxPack": "justify",
          "WebkitFlexDirection": "row",
          "WebkitFlexWrap": "wrap",
          "WebkitJustifyContent": "space-between",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "msFlexDirection": "row",
          "msFlexPack": "justify",
          "msFlexWrap": "wrap",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(0,0,0,1.00)",
            "fontFamily": "HelveticaNeue",
            "fontSize": "0.875rem",
            "lineHeight": "1.2249999999999999rem",
            "marginBottom": "0.25rem",
          }
        }
      >
        Account Balance
      </div>
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(128,128,128,1.00)",
            "fontFamily": "HelveticaNeue",
            "fontSize": "0.75rem",
            "lineHeight": "1.0499999999999998rem",
          }
        }
      >
        0x6273...Ef57
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "1rem",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitAlignItems": "center",
            "WebkitBoxAlign": "center",
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitFlexDirection": "row",
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "marginRight": "0.75rem",
            "msFlexAlign": "center",
            "msFlexDirection": "row",
            "position": "relative",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitFlexBasis": "auto",
              "bottom": "0px",
              "display": "flex",
              "flexBasis": "auto",
              "height": "2.5rem",
              "msFlexPreferredSize": "auto",
              "overflowX": "hidden",
              "overflowY": "hidden",
              "position": "relative",
              "width": "2.625rem",
              "zIndex": 0,
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "backgroundImage": "url(\\"relevantcoin.png\\")",
                "backgroundPosition": "center",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
                "bottom": "0px",
                "height": "100%",
                "left": "0px",
                "position": "absolute",
                "right": "0px",
                "top": "0px",
                "width": "100%",
                "zIndex": -1,
              }
            }
          />
          <img
            alt=""
            className="css-accessibilityImage-9pa8cd"
            draggable={false}
            src="relevantcoin.png"
          />
        </div>
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "color": "rgba(0,0,0,1.00)",
              "display": "flex",
              "fontFamily": "HelveticaNeue-CondensedBold",
              "fontSize": "2.25rem",
              "lineHeight": "2.5rem",
              "marginLeft": "0.625rem",
              "position": "relative",
            }
          }
        >
          
          111
          
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitBoxDirection": "normal",
        "WebkitBoxLines": "multiple",
        "WebkitBoxOrient": "horizontal",
        "WebkitFlexDirection": "row",
        "WebkitFlexWrap": "wrap",
        "borderBottomColor": "rgba(222,222,222,1.00)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "1px",
        "borderLeftColor": "rgba(222,222,222,1.00)",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "1px",
        "borderRightColor": "rgba(222,222,222,1.00)",
        "borderRightStyle": "solid",
        "borderRightWidth": "1px",
        "borderTopColor": "rgba(222,222,222,1.00)",
        "borderTopStyle": "solid",
        "borderTopWidth": "1px",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "msFlexDirection": "row",
        "msFlexWrap": "wrap",
        "paddingBottom": "1rem",
        "paddingLeft": "1rem",
        "paddingRight": "1rem",
        "paddingTop": "1rem",
        "position": "relative",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        data-event-off="click"
        data-for="mainTooltip"
        data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"You can transfer these coins to your Ethereum wallet.\\"}}"
        dir="auto"
        style={
          Object {
            "bottom": "0px",
            "left": "0px",
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "zIndex": 1,
          }
        }
      />
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(128,128,128,1.00)",
            "fontFamily": "HelveticaNeue",
            "fontSize": "0.75rem",
            "lineHeight": "1.0499999999999998rem",
            "marginRight": "1rem",
          }
        }
      >
        Unclaimed Coins
        : 
        0
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        data-event-off="click"
        data-for="mainTooltip"
        data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"These are the coins located in your connected Ethereum wallet.\\"}}"
        dir="auto"
        style={
          Object {
            "bottom": "0px",
            "left": "0px",
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "zIndex": 1,
          }
        }
      />
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(128,128,128,1.00)",
            "fontFamily": "HelveticaNeue",
            "fontSize": "0.75rem",
            "lineHeight": "1.0499999999999998rem",
            "marginRight": "1rem",
          }
        }
      >
        Metamask Coins
        : 
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        data-event-off="click"
        data-for="mainTooltip"
        data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"These are coins you got for referrals and verifying social accounts.\\\\nYou cannot transfer these coins to Metamask.\\"}}"
        dir="auto"
        style={
          Object {
            "bottom": "0px",
            "left": "0px",
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "zIndex": 1,
          }
        }
      />
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(128,128,128,1.00)",
            "fontFamily": "HelveticaNeue",
            "fontSize": "0.75rem",
            "lineHeight": "1.0499999999999998rem",
            "marginRight": "1rem",
          }
        }
      >
        Airdrop Coins
        : 
        111
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        data-event-off="click"
        data-for="mainTooltip"
        data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"These are coins that you are currently betting on posts.\\\\nThey get unlocked once the bets expire.\\"}}"
        dir="auto"
        style={
          Object {
            "bottom": "0px",
            "left": "0px",
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "zIndex": 1,
          }
        }
      />
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(128,128,128,1.00)",
            "fontFamily": "HelveticaNeue",
            "fontSize": "0.75rem",
            "lineHeight": "1.0499999999999998rem",
            "marginRight": "1rem",
          }
        }
      >
        Locked Tokens Coins
        : 
        0
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "position": "relative",
        }
      }
    >
      <div
        className="css-text-901oao"
        data-event-off="click"
        data-for="mainTooltip"
        data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"This is the ratio between unlocked and locked coins.\\"}}"
        dir="auto"
        style={
          Object {
            "bottom": "0px",
            "left": "0px",
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "absolute",
            "right": "0px",
            "top": "0px",
            "zIndex": 1,
          }
        }
      />
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "color": "rgba(128,128,128,1.00)",
            "fontFamily": "HelveticaNeue",
            "fontSize": "0.75rem",
            "lineHeight": "1.0499999999999998rem",
            "marginRight": "1rem",
          }
        }
      >
        Staking Power
        : 
        100%
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitBoxDirection": "normal",
        "WebkitBoxLines": "multiple",
        "WebkitBoxOrient": "horizontal",
        "WebkitFlexDirection": "row",
        "WebkitFlexWrap": "wrap",
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": "1rem",
        "msFlexDirection": "row",
        "msFlexWrap": "wrap",
        "position": "relative",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginRight": "1rem",
          "marginTop": "0.5rem",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        data-focusable={true}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "MozTransitionProperty": "opacity",
            "MozUserSelect": "none",
            "WebkitTransitionDuration": "0.15s",
            "WebkitTransitionProperty": "opacity",
            "WebkitUserSelect": "none",
            "cursor": "pointer",
            "msTouchAction": "manipulation",
            "msUserSelect": "none",
            "opacity": 1,
            "touchAction": "manipulation",
            "transitionDuration": "0.15s",
            "transitionProperty": "opacity",
            "userSelect": "none",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
              "WebkitUserSelect": "none",
              "color": "rgba(0,0,255,1.00)",
              "cursor": "pointer",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": "0.75rem",
              "lineHeight": "1.0499999999999998rem",
              "marginRight": "0.25rem",
              "msUserSelect": "none",
              "textDecorationColor": "rgba(0,0,255,1.00)",
              "userSelect": "none",
            }
          }
        >
          Claim Tokens
        </div>
      </div>
      <div
        className="css-text-901oao"
        dir="auto"
        style={
          Object {
            "display": "flex",
            "position": "relative",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitFlexBasis": "auto",
              "display": "flex",
              "flexBasis": "auto",
              "height": "0.85rem",
              "msFlexPreferredSize": "auto",
              "overflowX": "hidden",
              "overflowY": "hidden",
              "position": "relative",
              "width": "0.85rem",
              "zIndex": 0,
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "backgroundImage": "url(\\"info.png\\")",
                "backgroundPosition": "center",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
                "bottom": "0px",
                "display": "inline-flex",
                "height": "100%",
                "left": "0px",
                "position": "absolute",
                "right": "0px",
                "top": "0px",
                "width": "100%",
                "zIndex": -1,
              }
            }
          />
          <img
            alt=""
            className="css-accessibilityImage-9pa8cd"
            draggable={false}
            src="info.png"
          />
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginRight": "1rem",
          "marginTop": "0.5rem",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        data-event-off="click"
        data-for="mainTooltip"
        data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"If you have Relevant tokens in your Metamask wallet, you need to connect your account to be able to use them.\\"}}"
        style={
          Object {
            "display": "flex",
            "position": "relative",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozTransitionProperty": "opacity",
              "MozUserSelect": "none",
              "WebkitTransitionDuration": "0.15s",
              "WebkitTransitionProperty": "opacity",
              "WebkitUserSelect": "none",
              "cursor": "pointer",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "opacity": 1,
              "touchAction": "manipulation",
              "transitionDuration": "0.15s",
              "transitionProperty": "opacity",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "MozUserSelect": "none",
                "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
                "WebkitUserSelect": "none",
                "color": "rgba(0,0,255,1.00)",
                "cursor": "pointer",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": "0.75rem",
                "lineHeight": "1.0499999999999998rem",
                "msUserSelect": "none",
                "textDecorationColor": "rgba(0,0,255,1.00)",
                "userSelect": "none",
              }
            }
          >
            Connect Your Wallet
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginRight": "1rem",
          "marginTop": "0.5rem",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "position": "relative",
        }
      }
    >
      <a
        className="c0 ulink-mock"
        href="https://uniswap.exchange/swap?theme=dark&outputCurrency=0xdood"
        onClick={[Function]}
        target="_blank"
      >
        <div
          className="css-text-901oao"
          data-focusable={true}
          dir="auto"
          onClick={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "MozUserSelect": "none",
              "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
              "WebkitUserSelect": "none",
              "color": "rgba(0,0,255,1.00)",
              "cursor": "pointer",
              "fontFamily": "HelveticaNeue-Medium",
              "fontSize": "0.75rem",
              "lineHeight": "1.0499999999999998rem",
              "msUserSelect": "none",
              "textDecorationColor": "rgba(0,0,255,1.00)",
              "userSelect": "none",
            }
          }
          tabIndex="0"
        >
          Get more coins
        </div>
      </a>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginTop": "0.5rem",
          "msFlexAlign": "center",
          "msFlexDirection": "row",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        data-event-off="click"
        data-for="mainTooltip"
        data-tip="{\\"type\\":\\"TEXT\\",\\"props\\":{\\"text\\":\\"When you upvote posts you also bet coins on them. How much you bet can be decided automatically or manualy.\\"}}"
        style={
          Object {
            "display": "flex",
            "position": "relative",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "MozTransitionProperty": "opacity",
              "MozUserSelect": "none",
              "WebkitTransitionDuration": "0.15s",
              "WebkitTransitionProperty": "opacity",
              "WebkitUserSelect": "none",
              "cursor": "pointer",
              "msTouchAction": "manipulation",
              "msUserSelect": "none",
              "opacity": 1,
              "touchAction": "manipulation",
              "transitionDuration": "0.15s",
              "transitionProperty": "opacity",
              "userSelect": "none",
              "zIndex": 1,
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "MozUserSelect": "none",
                "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
                "WebkitUserSelect": "none",
                "color": "rgba(0,0,255,1.00)",
                "cursor": "pointer",
                "fontFamily": "HelveticaNeue-Medium",
                "fontSize": "0.75rem",
                "lineHeight": "1.0499999999999998rem",
                "msUserSelect": "none",
                "textDecorationColor": "rgba(0,0,255,1.00)",
                "userSelect": "none",
              }
            }
          >
            Disable
             Betting Mode
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-text-901oao"
    dir="auto"
    style={
      Object {
        "color": "rgba(0,0,0,1.00)",
        "fontFamily": "HelveticaNeue-Bold",
        "fontSize": "1.5rem",
        "lineHeight": "2.0999999999999996rem",
        "marginTop": "4.5rem",
      }
    }
  >
    Recent Activity
  </div>
  <div
    className="css-text-901oao"
    dir="auto"
    style={
      Object {
        "color": "rgba(0,0,0,1.00)",
        "fontFamily": "HelveticaNeue",
        "fontSize": "0.875rem",
        "lineHeight": "1.2249999999999999rem",
        "marginTop": "1rem",
      }
    }
  >
    Your rewards for upvoting links and discussion threads that are relevant to the community.
  </div>
</div>
`;
